api/base/serializers.py

```diff
@@ -1003,7 +1009,19 @@ def is_valid(self, clean_html=True, **kwargs):
         return ret
 
 
-class JSONAPISerializer(ser.Serializer):
+class PrefetchRelationshipsSerializer(ser.Serializer):
+    relationship_field_names = list()
+
+    def __init__(self, *args, **kwargs):
+        super(PrefetchRelationshipsSerializer, self).__init__(*args, **kwargs)
+        for field_name, field in self.fields.iteritems():
+            if field.source == '*':
+                self.relationship_field_names.append(field_name)
```

Comment:
@sloria I think `self.fields` is the subset of fields on the serializer. There's probably a fair about of relations that aren't exposed via the API that we wouldn't want to join/prefetch.